pkg/otlp/internal/serializerexporter/consumer.go

```diff
@@ -25,15 +25,16 @@ var _ translator.Consumer = (*serializerConsumer)(nil)
 
 type serializerConsumer struct {
 	cardinality collectors.TagCardinality
+	extraTags   []string
 	series      metrics.Series
 	sketches    metrics.SketchSeriesList
 }
 
 // enrichedTags of a given dimension.
 // In the OTLP pipeline, 'contexts' are kept within the translator and function differently than DogStatsD/check metrics.
 func (c *serializerConsumer) enrichedTags(dimensions *translator.Dimensions) []string {
-	enrichedTags := make([]string, 0, len(dimensions.Tags()))
-	enrichedTags = append(enrichedTags, dimensions.Tags()...)
+	enrichedTags := make([]string, 0, len(c.extraTags)+len(dimensions.Tags()))
+	enrichedTags = append(c.extraTags, dimensions.Tags()...)
```

Comment:
Tests are failing with:
```
/go/src/github.com/hDUx5Ydi/0/DataDog/datadog-agent/pkg/otlp/internal/serializerexporter/consumer.go:36:2: ineffectual assignment to enrichedTags
```

Because this should be:

```suggestion
	enrichedTags := make([]string, 0, len(c.extraTags)+len(dimensions.Tags()))
	enrichedTags = append(enrichedTags, c.extraTags...)
	enrichedTags = append(enrichedTags, dimensions.Tags()...)
```

We should not append to `c.extraTags` since this would modify it for all other callers of `enrichedTags`